### PR TITLE
Catch oath2 request issues.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/frontend/oauth2_service.dart [hunk 1]
```diff
@@ -36,9 +40,11 @@ class OAuth2Service {
     try {
       final api = new oauth2_v2.Oauth2Api(authClient);
       info = await api.userinfo.get();
+    } catch (e, st) {
+      _logger.log(Level.INFO, 'Bearer token lookup failed.', e, st);
```
Comment (sortie) [inline app/lib/frontend/oauth2_service.dart]: Can we catch a known exception, such as access denied? Then we can log that as info, while unknown exceptions can be logged as errors / warnings?

File: app/lib/frontend/oauth2_service.dart [hunk 2]
```diff
@@ -36,9 +40,11 @@ class OAuth2Service {
...
     } finally {
       authClient.close();
     }
-    return info.email;
+    return info?.email;
```
Comment (sortie) [inline app/lib/frontend/oauth2_service.dart]: This could be moved into the try block, and the catch clause could return null.

**Comment by isoos** [comment]
PTAL

File: app/lib/frontend/oauth2_service.dart [hunk 3]
```diff
@@ -36,9 +40,15 @@ class OAuth2Service {
     try {
       final api = new oauth2_v2.Oauth2Api(authClient);
       info = await api.userinfo.get();
+      return info.email;
+    } on AccessDeniedException catch (e, st) {
+      _logger.log(
+          Level.INFO, 'Bearer token lookup failed with access denied.', e, st);
```
Comment (sortie) [inline app/lib/frontend/oauth2_service.dart]: I would rephrase that as 'Access denied for bearer token'. What's a bearer token btw, maybe mention oauth or something?
